src/base/system.h

```diff
@@ -1831,6 +1831,22 @@ int str_utf16le_encode(char *ptr, int chr);
 */
 int str_utf8_check(const char *str);
 
+/*
+	Function: str_tokenize
+		Tokenizes a string.
```

Comment:
This doesn't tell me how to use the function.

```suggestion
		Split a string by a specified delimiter, returning the next token.

		On the first call, pass the `str` you want to split and a (potentially uninitialized) buffer for the state pointer of this function. The first token will be written (null-terminated, truncated to `bufsize` characters) to `buf`.
		
		On subsequent calls, pass the same parameters to `delim` and `state` and pass a null pointer to `str` to get the next tokens.
```

I think this complicated description shows that the function would be better off not taking `state` but rather just fetch one token and either return the remainder of the string or write it back to the input pointer. So either `str_next_token(const char **str, …)` or `const char *str_next_token(const char *str, …)`.

This would remove the special casing of the first case in the usages as well, replacing a different first case and a do-while-loop with a simpler while loop.